### DISCUSSION TITLE
modrinth ads (Discussion)

### DISCUSSION THREAD

**Discussion by sylv256** [uBlockOrigin/uAssets#24177] [General]
Discussion for #24171 
Read the EasyList policies before commenting. UBO adheres to these. https://easylist.to/pages/policy.html

**Comment by trunksbomb** [comment]
> Anything that leads to third-party(outside the website)

I'm sorry to be pedantic about it, but for the sake of clarity - would this include blocking an inline link like [Save 25% on GitHub with code XYZ](https://microsoft.com)? That's the last bit I'm unclear about. Thanks for taking the time to answer.

_Edit to add_: based on the provided link to the EasyList, this type of inline link on its own should be blocked. Separately, the image sans link would be blocked, too

**Reply by TelepathicGrunt** [reply]
This is the major part I have problems with. The rule there is so vague and broad it covers EVERYTHING. 

Modrinth provides me areas to link external sites for wiki, discord, source code, and donations. All of these link to outside Modrinth. Is unlock doing to block all these and make my modpack literally unusable for uBlock user? 
![IMG_0990](https://github.com/uBlockOrigin/uAssets/assets/40846040/1d337992-bb57-4ab0-8dd2-fa66412ca5d5)

If these links aren’t the intended ones? Then uBlock needs to be a LOT more clear of exactly what they count as something to block and what isn’t. As far as I see it, uBlock is ruling towards harming user experience rather than helping users at this rate

**Reply by sylv256** [reply]
This is simply disinformation. https://easylist.to/pages/policy.html uBlock Origin uses EasyList's policies. The rules are quite clear.

**Reply by TelepathicGrunt** [reply]
If uBlock blocks my issue tracker link too, I will definitely add text to tell users to disable uBlock on my mod page to report issues and see wiki. I will not tolerate uBlock blocking issue reporting harder for users for me. I live and die by issue reports. It’s how I make my mods better and the only ways I get bug reports

**Reply by RemagOfficial** [reply]
make the issue tracker an image with a link then report it to ublock when it gets blocked

**Reply by sylv256** [reply]
Issue tracker links are not ads, so they are not blocked.

**Reply by Jamalam360** [reply]
@sylv256 we're referring to this comment on the issue: 'Anything that leads to third-party(outside the website)'. I imagine it's just poor wording.

**Reply by TelepathicGrunt** [reply]
> This is simply disinformation.

If it was disinformation, then why was the opposite stated here? Communicating what is allowed or not should be clear up front. Here it was stated all third party links. 
https://github.com/uBlockOrigin/uAssets/issues/24171#issuecomment-2179238291

As long as my Modrinth allowed links are ok, I am cool.

**Reply by trunksbomb** [reply]
According to EasyList, uBlock _will_ block an inline affiliate link:
> Affiliate links/images/widgets

Thanks for clarifying.

**Reply by MrSoulPenguin** [reply]
Can't really blame people for thinking such a thing could occur. When it's said that the intended target could be ["Anything that leads to third-party(outside the website)"](https://github.com/uBlockOrigin/uAssets/issues/24171#issuecomment-2179238291)

**Reply by trunksbomb** [reply]
@TelepathicGrunt 

> As long as my Modrinth allowed links are ok, I am cool.

It appears that your social links and issue tracker link are fine, but I'm uncertain about links to Patreon and other services of the like - I think EasyList could probably go for some clarification on links to services like Patreon just to be specific since that's (IMO) kind of a gray area with the current wording. Although it may be covered by the opening paragraph "..doesn’t get commercially compensated for by third parties...".

**Comment by uBlock-user** [comment]
This will still be moderated, but the flow can be separated through individual threads.

**Comment by Jamalam360** [comment]
Asking here because the threads I try to reply too keep getting deleted - who exactly is this 'team'? /gen

What I have seen so far is modders, many I know and respect, expressing resentment towards what ublock is doing (with many valid reasons).

**Reply by sylv256** [reply]
The Easylist policies are not up for debate.

**Reply by Jamalam360** [reply]
Hence I'm not debating them...?

I don't think it's overly unreasonable for an ad blocker to block ads. It just seems like this situation has been handled poorly and the "moronic team" comment is just stoking the fire for no reason, especially when it seems, to me, that it's referring to any modder with a problem.

**Reply by Jamalam360** [reply]
(I see both sides of the debate, to be clear)

**Reply by garry-ut99** [reply]
> sylve256 : Discussion for #24171
> Jamalam360 : Asking here because the threads I try to reply too keep getting deleted

Also worth to mention another important thread:
- https://github.com/uBlockOrigin/uAssets/issues/24130

**Comment by RemagOfficial** [comment]
I have no idea why this was readded, it was reverted, everyone was happy that a dumb change was fixed and like half a day later it's still being discussed.

as a mod developer we shouldn't have to tell people to disable their adblocker just to not break our descriptions

**Reply by sylv256** [reply]
Breakages can be reported and will be fixed. Blocking ends where <u>legitimate</u> non-ad content begins.

**Comment by CammiePone** [comment]


**Reply by sylv256** [reply]
An adblocker blocks ads. If you have a problem, tell users to stop using an adblocker.

**Comment by sylv256** [comment]
@uBlock-user Is there a private place where I can report legitimate issues and updates without being dogpiled by bad-faith spammers?

**Reply by uBlock-user** [reply]
You will need to join the uAssets team, @gwarser

**Comment by sylv256** [comment]
Users are now using `github.io` HTML redirects. Is there any way to detect this?
![image](https://github.com/uBlockOrigin/uAssets/assets/18603248/ce38902a-df48-42a5-96b9-94a2b5b6a7a5)
![image](https://github.com/uBlockOrigin/uAssets/assets/18603248/7f241d5d-3905-4ebe-a37f-dd46bd8dc072)

**Reply by ghajini** [reply]
which link?

**Reply by sylv256** [reply]
`https://modrinth.com/mod/more-furnace-variants-lieonlion`

**Reply by ghajini** [reply]
fixable
`||raw.githubusercontent.com/LieOnLion/LieOnLion/main/images/banners/$image,domain=modrinth.com`

even if they use link redirects etc, ads can be blocked

**Reply by Jamalam360** [reply]
This endless game of cat and mouse is stupid (_from both parties_) and won't lead anywhere :laughing: 

edit: especially if the only solution is to block individual redirects lol

**Reply by Jamalam360** [reply]
@ghajini I'm no ublock expert (not even familiar with the syntax of blocks at all, tbh), but that looks brittle - this author in particular is using the `images/banner` directory to store other, non-ad images as well - example https://github.com/LieOnLion/LieOnLion/blob/main/images/banners/mcv/vanillaChests.png

Even if the block doesn't match that image because it's in a subdirectory, the author could innocently start using the main directory for non-ad images at any time.

**Reply by ghajini** [reply]
there are many ways to block ad

**Reply by Jamalam360** [reply]
I have no idea what that means, to be quite honest. As far as I can tell the filter you just committed will break the mod page that uses the non-ad banner in the banners directory on that GitHub

**Reply by ghajini** [reply]
if there's breakage, we will fix

**Reply by Jamalam360** [reply]
Ok. 

There is a breakage. The filter you use will incorrectly block images in the banners folder which are not an ad.

**Reply by Jamalam360** [reply]
@ghajini here is what i am referring to

without ublock:

![image](https://github.com/uBlockOrigin/uAssets/assets/56727311/bb30132a-e791-4407-8bd5-3c5c6ae2c89c)

with:

![image](https://github.com/uBlockOrigin/uAssets/assets/56727311/760f6c67-fa27-4fa9-8be0-1dd45b5544c7)

your filter incorrectly blocks https://github.com/LieOnLion/LieOnLion/blob/main/images/banners/mcv/vanillaChests.png, as I said. 

Creating individual filters like this is setting a **dangerous precedent** for accidentally breaking **legitimate non-ad content** in descriptions.

> Unless you plan to maintain an ever-updating and changing list of thousands of mod/plugin pages manually, this won't work.

**Reply by stephenhawk8054** [reply]
If the new filters cause breakages, any volunteers who's willing to fix will continue fixing it. 

If a website deliberately circumvents our filters to inject affiliate images, that website is not considered as normal website for us to re-consider for the performance issue or breakages any more. This is not the first website with that behavior. We are familiar with these kinds of websites so no need for unnecessary explanations.

**Reply by ghajini** [reply]
you mod guys don't even provided a link to test for breakages, just doing unnecessary chit-chat

**Reply by uBlock-user** [reply]
`https://modrinth.com/mod/more-chest-variants-lieonlion`

**Reply by Jamalam360** [reply]
First of all, @stephenhawk8054:

This isn't someone trying to work around the blocks. This is a problem with the filter. I tried to warn about the filter multiple times before it was implemented but nothing came of that. They haven't put that image in the banners directory to circumvent the blocks or to make it harder for ublock, it was already there.

@ghajini sorry, I didn't provide a link since this page was one that had been mentioned lots of times previously. I would appreciate it if you stopped clumping us all together as 'you mod guys' (or the 'moronic team' for that matter).

I accept that ublock is an ad blocker that blocks ads, but I don't think it should be pursuing filters that break regular content, with little interest in preventing those filters from being added (as I said, I attempted to warn about the problem multiple times).

It's also unrealistic to block the personal asset GitHub repositories of every modder! This breakage was bound to happen: the only way to prevent it would have been to look into LieOnLions asset repository before making the block - are you going to do that for every single sponsored modder?

I'm not trying to stop ublock blocking ads, but I don't think anyone wants it to be breaking websites.

**Reply by stephenhawk8054** [reply]
That's why I said

> If the new filters cause breakages, any volunteers who's willing to fix will continue fixing it.

The previous filters we used try to target specific affiliate ads and minimize false positives. When that page updates to workaround the filters and users report the ads re-appear, the filters would need to be updated again. And everytime the filters are updated, the chance of false positives increase. This is unavoidable, and would not happen if the website does not deliberately modify the page to avoid those first filters.

I had updated the filters to match the reports again hours ago: https://github.com/uBlockOrigin/uAssets/commit/1a9f584c4cceff791e10ed666192880e0a313361

The `/link-redirect` is the attempt to not link the destination link directly to bypass the previous filters. We all know that, you all know that, no need to play pretending here. If you see breakages, just report the exact links. It's just we can't consider this site as normal websites that all of the breakages are due to uBO's filters alone, and all of the extra explanations that are irrelevant to the breakages details are unnecessary.

---

Update: the content of the redirect link also specifically confirms the workaround

![image](https://github.com/uBlockOrigin/uAssets/assets/66517106/4634cb11-c184-49ba-afc4-5a14c3996753)

**Reply by Jamalam360** [reply]
To be clear, it's not 'the website' trying to avoid the filters - you're blocking user generated content. Modrinth itself has nothing to do with this.

Yes, some of my peers are desperately attempting to play an endless game of cat and mouse with you - not very productive. My issue is I tried to warn about the issue 3 times, twice before it was even committed, and it was seemingly ignored.

**Reply by stephenhawk8054** [reply]
When you report the issue, give the exact link where the issue occured, or else no one can help and the issue had to be ignored. I searched above and could not figure out which exact link does the issue occur. 

I had to look at the screenshot and guessed the link based on the content before updating the filter. Everyone here follows that minimum steps when reporting issue.

And yes, we know about all of the tweets about uBO from you, so again, no need to play any pretending here. Just report the breakages with details, that's all.

**Reply by Jamalam360** [reply]
I presume by tweets you mean the Mastodon account I set up less than an hour ago so I had somewhere to write about this. How you know about that I don't know. I can only assume you've been scouring the internet for information on me, which is quite frankly disturbing.

I'm not engaging in this anymore after that. I'll just implore you to actually consider what you are doing before implementing blocks. The previous block on LieOnLions repository was careless. A 20 second check would have told you that it broke other pages.

**Reply by dportvine** [reply]
@Jamalam360 
> I accept that ublock is an ad blocker that blocks ads

`uBlock Origin is not an "ad blocker", it's a wide-spectrum content blocker with CPU and memory efficiency as a primary feature.`

**Reply by Jamalam360** [reply]
Don't tag me, it automatically resubscribes me to notification when I am trying to leave. 

Besides, that distinction makes literally no difference to what I said, it's just pedantry.

**Reply by stephenhawk8054** [reply]
I said "you" here as the modrinth community, and modrinth website themselves who started this effort and talked about uBO: https://x.com/modrinth/status/1803169575061598525 (so don't say that modrinth itself has nothing to do with this).

From the beginning, there's no breakage reports that contain proper details about what are being blocked incorrectly. All just come and spam that the "affiliate images" are not ads. There are clear differences between sponsors put at a project which contain no affiliate codes, and the images with affiliate codes in the destination links we all have seen there, and it totally fits EasyList policies. Everyone can see that difference. "User's content" does not mean it can't contain ads or affiliate links.

uBO does what it does based on the policies that it has sticked to from the start of the project. And in fact, those filters would just end up in EasyList, since it's not for uBO only.

**Reply by uBlock-user** [reply]
> Besides, that distinction makes literally no difference to what I said, it's just pedantry.

Then DO NOT ever whine/bitch/moan and complain about uBO blocking user-generated content(which you and your moronic team did 2 days ago). uBO is a content-blocker. Expect content to be blocked if anything which is anti-user as per Easylist/EasyPrivacy's policy is found.

**Reply by ferriarnus** [reply]
As for what is broken, see https://github.com/uBlockOrigin/uAssets/discussions/24177#discussioncomment-9821526 . The image containing his chests is gone.

For the modrinth tweet, this is not an attack, but simply asking people to turn it of too support the modders on the site. I don't think asking this is wrong  as it is how many people are able to keep working on projects. They are also not taking steps against uBlock, or stopping users with adblockers from using the site.

**Reply by stephenhawk8054** [reply]
They said more before `asking people to turn it of too support the modders`. Don't just look at the end of their tweets.

As I said, if there are breakages, just report with proper steps and details. We also have a quick fix list to address emergency issue.

And when I said `From the beginning`, it means the issue threads created and spammed before this discussion thread being opened. And, as I said again, even with this thread, there's not a single link that any volunteers can know what is being hidden there. I literally looked at the screenshots and searched for the words appeared on that to get the link.

**Reply by Yuki2718** [reply]
BTW I have no idea why he keeps calling uBlockOrigin, uBlockOrigin, uBlockOrigin, ... (even referring [at]uBlockOrigin which declares `UNOFFICIAL fan-made mostly automated twitter`). Even before any such comments came, I clearly [commented](https://github.com/uBlockOrigin/uAssets/issues/24130#issuecomment-2173600287) that the only reason I didn't add to EL was EL doesn't support `nth-of-type`, AND at the same time added exactly the same filters to AdGuard. Then later added to EL which all major blocker use, as you all see. Ofc all the EL mantainers know these.

**Comment by stephenhawk8054** [comment]
I think the discussion can be closed if every side already talks their thoughts and does not want to drag this more. Affiliate links, users' contents, how uBO follows its policies, how the unnecessary things blowed up that affected proper reports... are said here: https://github.com/uBlockOrigin/uAssets/discussions/24177#discussioncomment-9826077

Future reports of ads and breakages will be addressed as normal if any volunteers have time for those issues. Any users who want to drag unnecessary things to future reports could be banned. If you want to discuss more about the policies, please do it at a more proper place with EasyList and other content blockers that use EasyList as well.
